pkg/virt-operator/creation/components/crds.go

```diff
@@ -37,6 +39,44 @@ const (
 	KUBEVIRT_PROMETHEUS_RULE_NAME = "prometheus-kubevirt-rules"
 )
 
+var (
+	VIRTUALMACHINE                   = "virtualmachines." + virtv1.VirtualMachineInstanceGroupVersionKind.Group
+	VIRTUALMACHINEINSTANCE           = "virtualmachineinstances." + virtv1.VirtualMachineInstanceGroupVersionKind.Group
+	VIRTUALMACHINEINSTANCEPRESET     = "virtualmachineinstancepresets." + virtv1.VirtualMachineInstancePresetGroupVersionKind.Group
+	VIRTUALMACHINEINSTANCEREPLICASET = "virtualmachineinstancereplicasets." + virtv1.VirtualMachineInstanceReplicaSetGroupVersionKind.Group
+	VIRTUALMACHINEINSTANCEMIGRATION  = "virtualmachineinstancemigrations." + virtv1.VirtualMachineInstanceMigrationGroupVersionKind.Group
+	KUBEVIRT                         = "kubevirts." + virtv1.KubeVirtGroupVersionKind.Group
+	VIRTUALMACHINESNAPSHOT           = "virtualmachinesnapshots." + snapshotv1.SchemeGroupVersion.Group
+	VIRTUALMACHINESNAPSHOTCONTENT    = "virtualmachinesnapshotcontents." + snapshotv1.SchemeGroupVersion.Group
+	PreserveUnknownFieldsFalse       = false
+)
+
+func patchValidation(crd *extv1beta1.CustomResourceDefinition) {
+	crd.Spec.PreserveUnknownFields = &PreserveUnknownFieldsFalse
+	validation, ok := CRDsValidation[crd.Spec.Names.Singular]
+	if !ok {
+		return
+	}
+	crvalidation := extv1beta1.CustomResourceValidation{}
+	k8syaml.NewYAMLToJSONDecoder(strings.NewReader(validation)).Decode(&crvalidation)
+	crd.Spec.Validation = &crvalidation
+	// workaround issues with metadata in spec
+	// TODO: find out if tests can be changed and this removed
```

Comment:
I think I know what's going on here. For now, keep this code. I'll follow up this and include you

Reply:
The problem here is that we have no control over this API. So I couldn't use `// +kubebuilder:pruning:PreserveUnknownFields` as in `VirtualMachineInstanceTemplateSpec`.
I think recommended fix would be to replace a `metav1.ObjectMeta` with `TemplateObjectMeta`. Reasoning behind this is that this embedded metadata doesn't need all attributes of `metav1.ObjectMeta`. I want to have a look to Kubernetes codebase to better understand how validation is handled.

Reply:
with #4261 you should be able to add the perserve unknown fields marker now to DataVolumeTemplateSpec just like with VirtualMachineInstanceTemplateSpec.